### DISCUSSION TITLE
Local user unable to run DAFOAM

### DISCUSSION THREAD

**Discussion by Karyan07** [mdolab/dafoam#72] [General]
HI,

I am a local user in this Linux system (CentOS7) and I do not have administrator privileges, while using OpenFOAM and DAFoam,
As a local user, when I run docker and when I try to run the Airfoil code given in the tutorial, the following error appears when running the preProcessing.sh script.

"./preProcessing.sh: line 11: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 12: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 13: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 14: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 15: logMeshGeneration.txt: Permission denied
"
and it says :

"
Generating mesh... Done!
cp: cannot create directory '0': Permission denied
"
Please tell me a way to fix this.

Thanks,
Karthik

**Comment by friedenhe** [comment]
If you have permission issues, try to open the docker container as usual, and copy the optimization run folder to `/home/dafoamuser` in the docker container and run jobs from there. You may need to change the run folder permission using `sudo chmod -R 777 your_optimization_folder`. Check [this page](https://dafoam.github.io/mydoc_get_started_run.html) for the sudo password for dafoamuser.

**Comment by Karyan07** [comment]
Hey friedenhe,

 Thank you for your reply. I would however want to run the files locally, instead of running it in the container, I had no problem doing it on my personal computer as I am the administrative user, However, when using it at the university, I am a local user and I do not have administrator privileges. Therefore, I would like to know what I should do to run it locally. 

Please see how I run the codes in the container in my personal computer (This is how I want to run it): 

^[[Akaryan07@karyan07-VirtualBox:~/OpenFOAM/karyan07-8/tutorials-master/NACA0012_Airfoil/incompressib
karyan07@karyan07-VirtualBox:~/OpenFOAM/karyan07-8/tutorials-master/NACA0012_Airfoil/incompressible$ docker run -it --rm -u dafoamuser --mount "type=bind,src=$(pwd),target=/home/dafoamuser/mount" -w /home/dafoamuser/mount dafoam/opt-packages:v2.1.3 bash
dafoamuser@a1cb1b1e6170:~/mount$ ./Allclean.sh
Delete everything and resume to the default setup (y/n)?y
Cleaning...
dafoamuser@a1cb1b1e6170:~/mount$ ./preProcessing.sh
Generating mesh..
Generating mesh.. Done! 

Thanks,
Karthik

**Reply by friedenhe** [reply]
@Karthi019 The trick mentioned above is the only option we have at this moment if you want to run the DAFoam docker image there. Actually, this issue is probably not because you don't have admin access. Instead, it is because the system has more than one user and our docker package configuration assumes you are the first user, who usually has the admin access. The docker image is designed for first time users who want to try DAFoam, and we suggest you build everything from scratch for production runs. See this [page](https://dafoam.github.io/mydoc_installation_source.html).

**Reply by Karyan07** [reply]
Thanks for your prompt reply. I'm not sure about what you mean by build everything from scratch for production runs? I saw that page and it looks like I've to install all the components separately onto my system. 

And for running it onto the container itself, could you please give me an example to copy the tutorial to the container? 

Let's say I want to run the NACA airfoil Incompressible case which is locally on /home/Openfoam/ ?

Your help will be much appreciated because I tried copying it as per the docker command documentation, but I keep getting errors. 

Thanks again

**Comment by friedenhe** [comment]
This could be a common issue. I just added it to FAQ, check the workaround from [here](https://dafoam.github.io/mydoc_get_started_faq.html#why-i-keep-getting-a-permission-denied-error-when-using-the-dafoam-docker-image).
